src/Phinx/Console/Command/Rollback.php

```diff
@@ -100,17 +107,71 @@ protected function execute(InputInterface $input, OutputInterface $output)
         if (isset($envOptions['name'])) {
             $output->writeln('<info>using database</info> ' . $envOptions['name']);
         }
+        
+        $versionOrder = $this->getConfig()->getVersionOrder();
+        $output->writeln('<info>ordering by </info>' . $versionOrder);
 
         // rollback the specified environment
-        $start = microtime(true);
-        if (null !== $date) {
-            $this->getManager()->rollbackToDateTime($environment, new \DateTime($date), $force);
+        if (null === $date) {
+            $targetMustMatchVersion = true;
+            $target = $version;
         } else {
-            $this->getManager()->rollback($environment, $version, $force);
+            $targetMustMatchVersion = false;
+            $target = $this->getTargetFromDate($date);
         }
+
+        $start = microtime(true);
+        $this->getManager()->rollback($environment, $target, $force, $targetMustMatchVersion);
         $end = microtime(true);
 
         $output->writeln('');
         $output->writeln('<comment>All Done. Took ' . sprintf('%.4fs', $end - $start) . '</comment>');
     }
+
+    /**
+     * Get Target from Date
+     *
+     * @param string $date The date to convert to a target.
+     * @return The target
+     */
+    public function getTargetFromDate($date)
+    {
+        if (!preg_match('/^\d{4,14}$/', $date)) {
+            throw new \InvalidArgumentException(sprintf('Invalid date. Format is YYYYmmddHHiiss (with no mandatory '.
+                'part, but the missing part(s) must all be in the end of the date string)', $date));
+        }
+
+        // what we need to append to the date according to the possible date string lengths
+        $dateStrlenToAppend = array(
+            14 => '',
+            12 => '00',
+            10 => '0000',
+            8 => '000000',
+            6 => '01000000',
+            4 => '0101000000',
+        );
+
+        $format = null;
+
+        foreach($dateStrlenToAppend as $dateStrLen => $append) {
+            if (strlen($date) === $dateStrLen) {
+                $target = $date . $append;
+                break;
+            }
+        }
+
+        if ($target === null) {
+            throw new \InvalidArgumentException(sprintf('Invalid date. Format is YYYYmmddHHiiss (with no mandatory '.
+                'parts, but the missing part(s) must all be in the end of the date string', $date));
+        }
+
+        $dateTime = \DateTime::createFromFormat('YmdHis', $target);
+
+        if ($dateTime === false) {
+            throw new \InvalidArgumentException(sprintf('Invalid date. Format is YYYYmmddHHiiss (with no mandatory '.
+                'parts, but the missing part(s) must all be in the end of the date string', $date));
+        }
+
+        return $dateTime->format('YmdHis');
```

Comment:
Line 168 makes a string, can't call method on a string. As all unit tests are passing, it would seem this code is not called